src/colvaratoms.cpp

```diff
@@ -808,24 +808,24 @@ int cvm::atom_group::create_sorted_ids()
 
   // Sort the internal IDs
   std::list<int> sorted_atoms_ids_list;
-  for (size_t i = 0; i < this->size(); i++) {
+  for (size_t i = 0; i < atoms_ids.size(); i++) {
```

Comment:
Note: these change are necessary to obtain atom IDs when groups are marked as scalable.